src/lib/optimizer/strategy/join_to_semi_join_rule.cpp

```diff
@@ -72,20 +74,30 @@ void JoinToSemiJoinRule::_apply_to_plan_without_subqueries(const std::shared_ptr
       }
 
       // Determine which node to use for Semi-Join-filtering and check for the required uniqueness guarantees.
-      if (*join_node->prunable_input_side() == LQPInputSide::Left &&
-          join_node->left_input()->has_matching_ucc(equals_predicate_expressions_left)) {
-        join_node->join_mode = JoinMode::Semi;
-        const auto temp = join_node->left_input();
-        join_node->set_left_input(join_node->right_input());
-        join_node->set_right_input(temp);
-      } else if (*join_node->prunable_input_side() == LQPInputSide::Right &&
-                 join_node->right_input()->has_matching_ucc(equals_predicate_expressions_right)) {
-        join_node->join_mode = JoinMode::Semi;
+      if (*join_node->prunable_input_side() == LQPInputSide::Left) {
+        auto matching_ucc = join_node->left_input()->get_matching_ucc(equals_predicate_expressions_left);
```

Comment:
Again, not sure if this can be const
```suggestion
        const auto matching_ucc = join_node->left_input()->get_matching_ucc(equals_predicate_expressions_left);
```

Reply:
(same below)